fvm/evm/stdlib/contract.go

```
@@ -38,6 +42,491 @@ var evmAddressBytesType = sema.NewConstantSizedType(nil, sema.UInt8Type, types.A
+					values = append(values, int16(value))
+					typ, err := gethABI.NewType("int16", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int32Value:
+					values = append(values, int32(value))
+					typ, err := gethABI.NewType("int32", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int64Value:
+					values = append(values, int64(value))
+					typ, err := gethABI.NewType("int64", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int128Value:
+					values = append(values, value.BigInt)
+					typ, err := gethABI.NewType("int128", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int256Value:
+					values = append(values, value.BigInt)
+					typ, err := gethABI.NewType("int256", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.BoolValue:
+					values = append(values, bool(value))
+					typ, err := gethABI.NewType("bool", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case *interpreter.CompositeValue:
+					if value.QualifiedIdentifier == "EVM.EVMAddress" {
+						bytes, err := interpreter.ByteArrayValueToByteSlice(
+							inter,
+							value.GetMember(inter, locationRange, evmAddressTypeBytesFieldName),
+							locationRange,
+						)
+						if err != nil {
+							panic(err)
+						}
+						values = append(values, gethCommon.Address(bytes))
+						typ, err := gethABI.NewType("address", "", nil)
+						if err != nil {
+							panic(err)
+						}
+						arguments = append(arguments, gethABI.Argument{Type: typ})
+					}
```

Report an error if the value is not an `EVM.EVMAddress` instead of silently ignoring the composite value